api/src/pcapi/routes/native/v1/authentication.py

```diff
@@ -207,3 +212,80 @@ def validate_email(body: ValidateEmailRequest) -> ValidateEmailResponse:
     )
 
     return response
+
+
+@blueprint.native_v1.route("/oauth/google/authorize", methods=["POST"])
+@spectree_serialize(
+    response_model=authentication.SigninResponse,
+    on_success_status=200,
+    on_error_statuses=[400, 401],
+    api=blueprint.api,
+)
+@ip_rate_limiter()
+@feature_required(FeatureToggle.WIP_ENABLE_GOOGLE_SSO)
+def google_auth(body: authentication.GoogleSigninRequest) -> authentication.SigninResponse:
+    # postmessage is needed when the app used a popup to share the auth code
+    # see https://stackoverflow.com/questions/71968377
+    redirect_uri = "postmessage"
+    token = oauth.google.fetch_access_token(code=body.code, redirect_uri=redirect_uri)
+    google_user = pydantic_v1.parse_obj_as(authentication.GoogleUser, oauth.google.parse_id_token(token))
+    email = google_user.email
+    sso_user_id = google_user.sub
+
+    single_sign_on = users_repo.get_single_sign_on("google", sso_user_id)
+    if not single_sign_on:
+        user = users_repo.find_user_by_email(email)
+    else:
+        user = single_sign_on.user
+
+    if not user:
```

Comment:
On ne crée pas le compte si on trouve rien ?
Sinon ça voudrait dire que le jeune doit créer une compte sur l'app classique, puis se connecter avec google pour que ça marche, c'est bizarre non ?

Reply:
En effet, la création de compte via SSO est un chantier qui arrive dans les prochaines itérations

Dans tous les cas, ça aurait fait beaucoup pour une seule PR